content/200-concepts/100-components/02-prisma-client/260-client-extensions/120-query.mdx

```diff
@@ -9,8 +9,186 @@ tocDepth: 3
+
 </TopBlock>
+
+## Enable the preview feature
+
+Before you create Prisma Client extensions, you must enable the `clientExtensions` feature flag in the `generator` block of your `schema.prisma` file, as follows:
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["clientExtensions"]
+}
+```
+
+## Create custom queries
+
+To create a custom query, use the following structure.
+
+```ts
+const xprisma = prisma.$extends({
+  name?: 'name',
+  query?: {
+    user: { ... } //in this case, we add a query to the `user` model
+  },
+});
+```
+
+- `name`: an optional field that you can use to name the extension. This name appears in error logs.
+
+In the following example, a custom query finds users who are older than 18 years:
+
+```ts
+const xprisma = prisma.$extends({
+  query: {
+    user: {
+      async findMany({ model, action, args, data }) {
+        args.where.age = { gt: 18 }
+        const myData = await data
+        return data
+      },
+    },
+  },
+})
+
+await xprisma.user.findMany() // returns users whose age is greater than 18
+```
+
+The `query` object contains functions that map to the names of the [Prisma Client operations](/docs/reference/api-reference/prisma-client-reference#model-queries), such as `findUnique`, `findFirst`, `findMany`, `count`, and `create`.
+
+In the above example, a call to `xprisma.user.findMany` triggers `query.findMany` on execution. Each callback receives a type-safe `{ model, operation, args, data }` object that describes a query as follows:
+
+- `model`: the property name of the model whose query is being extended.
+
+  In the above example, the `model` is a string of type `"User"`.
+
+- `operation`: the property name of the operation being extended and executed.
+
+  In the above example, the `operation` is a string of type `"findMany"`.
+
+- `args`: the property that carries the specific query input information to be extended.
+
+  This is a type-safe object that is meant to be mutated before the query happens. You can mutate any of the properties in `args`, except for `include` or `select`, because that would change the expected output type in such a way that it breaks type safety.
+
+- `data`: the property that is a promise of the result of the query that is being executed.
+
+  You can `await` and then mutate this promise, because its value is also type-safe. TypeScript catches any unsafe mutations on that object. Doing `await data` is equivalent to calling `next(...)` in a middleware. You are free to mutate `args` before `then` is triggered, which leads to a more natural way of writing such extensions. `args` is fully type-safe, meaning that some properties are `readonly`, (like `includes` or `select`).
+
+### Nested queries
```

Comment:
Nested queries have not been planned, we'd work on them if time allows.

Reply:
Thanks. I'll leave this comment in place to remind me to remove this section if we don't have this ready in time for release